src/helpers/update-tasks.ts

```diff
@@ -0,0 +1,172 @@
+    logger.info(`We are passed the deadline on ${issue.url}, should we send a reminder? ${!lastReminder}`);
+    if (!lastReminder && (await remindAssignees(context, issue))) {
+      await supabase.repositories.upsert({
+        url: issue.url,
+        deadline: deadline.toJSDate(),
+        lastReminder: now.toJSDate(),
+        lastCheck: now.toJSDate(),
+      });
+    }
+  }
+}
+
+async function updateReminders(context: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
+  const {
+    adapters: { supabase },
+    logger,
+    config,
+  } = context;
+  const now = DateTime.now();
+  const activity = await getAssigneesActivityForIssue(context, issue);
+  const deadline = DateTime.fromISO(issue.deadline);
+  const deadlineWithThreshold = deadline.plus({ day: config.unassignUserThreshold });
+  const reminderWithThreshold = deadline.plus({ day: config.sendRemindersThreshold });
+
+  if (activity?.length) {
+    const lastCheck = DateTime.fromISO(issue.last_check);
+    const timeDiff = now.diff(lastCheck);
+    const newDeadline = deadline.plus(timeDiff);
+    logger.info(
+      `Activity found on ${issue.url}, will move the deadline forward from ${deadline.toLocaleString(DateTime.DATETIME_MED)} to ${newDeadline.toLocaleString(DateTime.DATETIME_MED)}`
+    );
+    await supabase.repositories.upsert({ url: issue.url, deadline: newDeadline.toJSDate(), lastCheck: now.toJSDate() });
+  } else {
+    if (now >= deadlineWithThreshold) {
+      await unassignUserFromIssue(context, issue);
+    } else if (now >= reminderWithThreshold) {
+      await remindAssigneesForIssue(context, issue);
+    } else {
+      logger.info(
+        `Nothing to do for ${issue.url}, still within due-time (now: ${now.toLocaleString(DateTime.DATETIME_MED)}, reminder ${reminderWithThreshold.toLocaleString(DateTime.DATETIME_MED)}, deadline: ${deadlineWithThreshold.toLocaleString(DateTime.DATETIME_MED)})`
+      );
+    }
+  }
+}
+
+export async function updateTasks(context: Context) {
+  const {
+    adapters: { supabase },
+    logger,
+  } = context;
+  const watchedRepoList = await supabase.repositories.get();
+
+  if (!watchedRepoList?.length) {
+    logger.info("No watched repos have been found, no work to do.");
+    return false;
+  }
+  for (const watchedIssue of watchedRepoList) {
+    await updateReminders(context, watchedIssue);
+  }
+  return true;
+}
+
+async function getAssigneesActivityForIssue({ octokit, payload }: Context, issue: Database["public"]["Tables"]["repositories"]["Row"]) {
+  const { repo, owner, issue_number } = parseGitHubUrl(issue.url);
+  return octokit.paginate(
+    octokit.rest.issues.listEvents,
+    {
+      owner,
+      repo,
+      issue_number,
+      per_page: 100,
+    },
+    (res) =>
+      res.data.filter(
+        (o) =>
+          payload.issue?.assignees?.find((assignee) => assignee?.login === o.actor.login) &&
+          DateTime.fromISO(o.created_at) >= DateTime.fromISO(issue.last_check)
```

Comment:
personally I'd move the second part out of the function because the name of the function doesn't imply that it will also filter events since last check